src/harness/testcases/WINNF_FT_S_EXZ_testcase.py

```diff
@@ -0,0 +1,443 @@
+
+
+    # Register N2 devices
+    cbsd_ids_N2 = self.assertRegistered(config['registrationRequests_N2'])
+
+    # Register N3 devices
+    cbsd_ids_N3 = self.assertRegistered(config['registrationRequests_N3'])
+
+    # Register N4 devices
+    cbsd_ids_N4 = self.assertRegistered(config['registrationRequests_N4'])
+
+    # Execute CPAS and wait untill completion
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Generating grant requests for N2 and validating responses
+    grant_request = []
+    for i in range(len(config['grants_N2'])):
+        config['grants_N2'][i]['cbsdId'] = cbsd_ids_N2[i]
+        grant_request.append(config['grants_N2'][i])
+
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), len(grant_request))
+
+    for response_num, resp in enumerate(response):
+          self.assertEqual(resp['cbsdId'], cbsd_ids_N2[response_num])
+          self.assertTrue('grantId' in resp)
+          #self.assertEqual(resp['response']['responseCode'], 400)
+
+
+    # Generating grant requests for N3 and validating responses
+    grant_request = []
+    for i in range(len(config['grants_N3'])):
+        config['grants_N3'][i]['cbsdId'] = cbsd_ids_N3[i]
+        grant_request.append(config['grants_N3'][i])
+
+  
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), len(grant_request))
+
+    for response_num, resp in enumerate(response):
+          self.assertEqual(resp['cbsdId'], cbsd_ids_N3[response_num])
+          self.assertTrue('grantId' in resp)
+          #self.assertEqual(resp['response']['responseCode'], 400)
+
+    # Generating grant requests for N4 and validating responses
+    grant_request = []
+    for i in range(len(config['grants_N4'])):
+        config['grants_N4'][i]['cbsdId'] = cbsd_ids_N4[i]
+        grant_request.append(config['grants_N4'][i])
+
+
+    request = {'grantRequest': grant_request}
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), len(grant_request))
+
+    for response_num, resp in enumerate(response):
+          self.assertEqual(resp['cbsdId'], cbsd_ids_N4[response_num])
+          self.assertTrue('grantId' in resp)
+          #self.assertEqual(resp['response']['responseCode'], 400)
+
+
+  def generate_EXZ_2_default_config(self, filename):
+    """Generates the WinnForum configuration for EXZ.2."""
+    # Load device info
+    # Load Devices located outside 50 meters of exclusion zones
+    device_N1_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_N1_1['installationParam']['latitude'] = 40.31260
+    device_N1_1['installationParam']['longitude'] =  -96.25100
+    device_N1_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
```

Comment:
Same comment re: conditional registration data.

Reply:
Removed EXZ.2 from this PR. We will submit another PR once EXZ.1 get approved. We will code this similar to approved EXZ.1